src/utils/hooks/useLoadApplication.tsx

```diff
@@ -33,6 +36,12 @@ const useLoadApplication = (props: useLoadApplicationProps) => {
 
       const sections = getApplicationSections(application.applicationSections)
       setSections(sections)
+
+      setAppStatus({
+        stage: application?.stage,
+        status: application?.status,
+        outcome: application?.outcome,
+      })
```

Comment:
I was originally thinking that isLoading (isReady), would be a hook inside useLoadApplication. (rather then in ApplicationPage). What do you think ?

Reply:
So the same logic that I have here is in `useLoadApplication`, and ApplicationPage just gets its state from the returned `loading` value? Is that what you mean?

Reply:
Okay, have moved `isReady` into `useLoadApplication` (so it's only responsible for setting `skip` in that hook) and moved the `processRedirect` inside the `useEffect` in the ApplicationPage.

0d3b95b

Is that what you had in mind?

Reply:
Yp, but I was also thinking that it will be returned from useLoadApplication

Reply:
Was going to do that, but figured it could be determined by `application` and `loading`, no?

Reply:
yep, that's good. I was just wondering for use cases further down the track. I guess can always determine if application has been loaded by checking loading or application. No need to change.